### DISCUSSION TITLE
Capturing Terminal Output

### DISCUSSION THREAD

**Discussion by Dj-1112** [Microsoft/node-pty#624] [Q&A]
## Environment details

- OS: Linux
- OS version: ubuntu 20.04
- node-pty version: 0.10.1

## Issue description
Hi,
We have integrated a web-terminal into our web application, we are using node-pty in the back end and x-term on the front end.
The terminal is working as expected, but we need to capture the output that is coming from the terminal.
We are using web-sockets to send data back and forth, the problem is that everything i type is being sent back and forth and since the shell is also echoing the commands that is also being sent back and forth.
How do we just capture the output of any command that got executed? so that we can parse it and display a success message else where in the page.

**Comment by Tyriar** [comment]
Separating input and output in a shell is a non-trivial problem unless you spawn a new process like `bash -c script` every time.

**Reply by Dj-1112** [reply]
So that would mean that a shell script is running behind the scenes to check the output at all times along with the usual terminal spawned by node pty right?

**Reply by Tyriar** [reply]
If you're not launching multiple ptys you would do this either by regex checking the prompt (unreliable) or by adding some shell integration script that injects sequences that the terminal/client script can intercept and understand.

**Reply by Dj-1112** [reply]
But we are launching multiple pty processes by assigning the uuid for the sessions

**Reply by Tyriar** [reply]
I mean if you launch a pty per thing that is run (eg. `bash -c "echo hello world"`), all the data will be the terminal output.
